setup.sh

```diff
@@ -0,0 +1,283 @@
+    fi
+}
+
+is_zsh() {
+    if [ -n "$ZSH_VERSION" ]; then
+        # 0 = true
+        return 0
+    elif [ -n "$BASH_VERSION" ]; then
+        return 1
+    fi
+}
+
+#
+# Resets the color
+#
+msg_reset () {
+  RESET='\033[0m'
+  printf "${RESET}\n"
+}
+
+#
+# Log an error
+#
+error () {
+  RED_FORE='\033[0;31m'
+  printf "${RED_FORE}ERROR: $1"
+  msg_reset
+}
+
+#
+# Log verbose message
+#
+verbose () {
+  if [ $VERBOSE_OUTPUT -eq 1 ]; then
+    return
+  fi
+  CYAN_FORE='\033[0;36m'
+  printf "${CYAN_FORE}INFO: $1"
+  msg_reset
+}
+
+#
+# Log message
+#
+msg () {
+  printf "$1\n"
+}
+
+#
+# Install Homebrew
+#
+install_homebrew () {
+    msg "Installing Homebrew"
+
+    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+    brew tap caskroom/cask
+
+    if [ $? -ne 0 ]; then
+        error "Was not able to install Homebrew."
+        exit 1
+    fi
+}
+
+#
+# Install ruby envioronment for linux
+#
+install_rbenv_linux() {
+    msg "installing ruby"
+    sudo apt update
+    git clone git://github.com/sstephenson/rbenv.git ~/.rbenv
+    git clone git://github.com/sstephenson/ruby-build.git ~/.rbenv/plugins/ruby-build
+
+    sudo apt-get install -y libreadline-dev
+
+    msg "git repos cloned"
+
+    if [ -n "$ZSH_VERSION" ]; then
+        echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.zshrc
+        echo 'export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH"' >> ~/.zshrc
+        source ~/.zshrc
+    elif [ -n "$BASH_VERSION" ]; then
+        echo 'export PATH="$HOME/.rbenv/bin:$PATH"' >> ~/.bashrc
+        echo 'export PATH="$HOME/.rbenv/plugins/ruby-build/bin:$PATH"' >> ~/.bashrc
+        source ~/.bashrc
+    fi
+}
+
+# 
+# Install rbenv and ruby-build
+# 
+install_rbenv () {
+    echo '2.3.1' >> ~/.ruby_version
+    msg "Installing ruby-build and rbenv"
+
+    if isMac;
+    then
+        brew install ruby ruby-build rbenv
+    else 
+        install_rbenv_linux
+    fi
+    msg "installed ruby"
+
+    if [ -n "$ZSH_VERSION" ]; then
+        echo 'eval "$(rbenv init -)"' >> ~/.zshrc
+    elif [ -n "$BASH_VERSION" ]; then
+        echo 'eval "$(rbenv init -)"' >> ~/.bashrc
+    fi
+
+    msg "installing ruby 2.3.1, this will take a few minutes..."
+    CONFIGURE_OPTS="--disable-install-doc --enable-shared" rbenv install 2.3.1
+
+    if [ $? -ne 0 ]; then
+        error "Was not able to install rbenv, or rbenv was already installed, will continue with installation."
+    fi
+
+    verbose "installed ruby-build and rbenv"
+    rbenv global 2.3.1
+    eval "$(rbenv init -)"
+    rbenv rehash
+    source ~/.bashrc
+}
+
+# 
+# Install postgres
+# 
+install_postgres () {
+    msg "Installing postgres"
+
+    if isMac;
+    then
+        brew cask install postgres --appdir=/Applications
+
+        export PATH=/Applications/Postgres.app/Contents/Versions/*/bin:$PATH
+
+        msg "installed postgress, should now be on path"
+        # TODO replace this with cli open.
+        open -a postgres
```

Comment:
http://superuser.com/questions/442554/postgresql-homebrew-installation-lacks-config-files

Use `initdb` (this lives in the `bin` folder with `postgres`) and create a database in the DF `tmp` directory: `/path/to/doubtfire-api/tmp/postgres`. Then, set the database to the `tmp` directory. Will need to use `pwd` to get the absolute path of the `tmp` directory using the `-D` switch.